Update to indexByObj functionality

youngaj opened the pull request:
The existing indexByObj functionality depends on the image/photo object to have a unique id field.  This is not documented in the ReadMe but this change will allow the function to work without the "id" field.

youngaj committed:
Update to indexByObj functionality

The existing indexByObj functionality depends on the image/photo object to have a unique id field.  This is not documented in the ReadMe but this change will allow the function to work without the "id" field.

abuechler commented:
The ID field is generated by the directive if there is none as you can see [here](https://github.com/abuechler/angular-superbox/blob/master/src/angular-superbox.js#L30). That's why it's not mentioned in the README. But you're right if an id field is there, its expected to be unique across given images. Which is a sensible assumption IMHO.

abuechler reviewed (changes requested):
When using angular.equals() there is no more need to generate the ids. Therefore remove them as well.

youngaj committed:
Removing id generation given use of angular.equals

Updates based on PR feedback.

angular.equals() removes the need to generate an "id" field if there one is not present on the model.  Additionally if the model has an existing "id" field that field would need to be unique which can't be guaranteed.

youngaj commented:
@abuechler, made the requested changes and updated PR.
